superset/db_engine_specs/base.py

```diff
@@ -1057,30 +1057,33 @@ def extract_errors(
         ]
 
     @classmethod
-    def adjust_database_uri(  # pylint: disable=unused-argument
+    def adjust_engine_params(  # pylint: disable=unused-argument
         cls,
         uri: URL,
-        selected_schema: Optional[str],
-    ) -> URL:
+        connect_args: Dict[str, Any],
+        catalog: Optional[str],
+        schema: Optional[str],
```

Comment:
I noticed that in some of the class overrides, catalog and schema have kwargs with defaults. Do we want to be consistent?

Reply:
Ah, good point, let me add defaults here.